src/core/types.ts

```diff
@@ -151,20 +152,45 @@ export interface ExternalConnector<P extends IProvider = IProvider> {
   widget: WidgetComponent<P>;
 }
 
+export interface ContractParams {
+  // Manual type of the contract can be used by the wallets without
+  // automatic contract detection capabilities - Ledger
+  type?: SigningStep;
+  stakerPk: string;
+  covenantPks: string[];
+  finalityProviders: string[];
+  covenantThreshold: number;
+  minUnbondingTime: number;
+  stakingDuration: number;
+  magicBytes: string;
+}
+
+export interface Contract {
+  id: string;
+  params: ContractParams;
+}
```

Comment:
```suggestion
export interface Contract {
  id: string;
  params: Record<string, string | number | string[] | number[]>;
}
```

You need use more abstract type for contract

Reply:
Do we achieve a better result with a much more abstract `Record` stuff?